src/bit-systems/audio-zone-system.ts

```diff
@@ -0,0 +1,318 @@
+  if (!curr && !acc) return {} as AudioSettings;
+  else if (curr && !acc) return curr;
+  else if (!curr && acc) return acc;
+  else
+    return REDUCED_KEYS.reduce((result: ReducedAudioSettings, key: ReducedAudioSettingsKeys): AudioSettings => {
+      if (curr[key] !== undefined && acc[key] !== undefined) {
+        result[key] = Math.min(acc[key]!, curr[key]!);
+      } else if (curr[key] !== undefined && acc[key] === undefined) {
+        result[key] = curr[key];
+      } else if (curr[key] === undefined && acc[key] !== undefined) {
+        result[key] = acc[key];
+      }
+      return result;
+    }, {});
+};
+
+const addOrRemoveZone = (zones: Set<number>, zone: number, position: Vector3) => {
+  if (!zones) return;
+  const isInZone = isZoneEnabled(zone) && zoneContains(zone, position);
+  const wasInZone = zones.has(zone);
+  if (isInZone && !wasInZone) {
+    zones.add(zone);
+  } else if (!isInZone && wasInZone) {
+    zones.delete(zone);
+  }
+};
+
+const currZones = new Map<number, Set<number>>();
+const prevZones = new Map<number, Set<number>>();
+const aabbs = new Map<number, Box3>();
+
+const listenerPos = new Vector3();
+const audioZoneQuery = defineQuery([AudioZone]);
+const audioZoneEnterQuery = enterQuery(audioZoneQuery);
+const audioZoneExitQuery = exitQuery(audioZoneQuery);
+const audioZoneSourceQuery = defineQuery([AudioEmitter]);
+const audioZoneSourceEnterQuery = enterQuery(audioZoneSourceQuery);
+const audioZoneSourceExitQuery = exitQuery(audioZoneSourceQuery);
+const audioZoneListenerQuery = defineQuery([AudioListenerTag]);
+const audioZoneListenerEnterQuery = enterQuery(audioZoneListenerQuery);
+const audioZoneListenerExitQuery = exitQuery(audioZoneListenerQuery);
+
+export function audioZoneSystem(world: HubsWorld) {
+  const zones = audioZoneQuery(world);
+  if (!zones.length) return;
+
+  [...audioZoneSourceEnterQuery(world), ...audioZoneListenerEnterQuery(world)].forEach(entity => {
+    currZones.set(entity, new Set());
+    prevZones.set(entity, new Set());
+  });
+  [...audioZoneSourceExitQuery(world), ...audioZoneListenerExitQuery(world)].forEach(entity => {
+    currZones.delete(entity);
+    prevZones.delete(entity);
+  });
+
+  audioZoneEnterQuery(world).forEach(zone => {
+    const obj = APP.world.eid2obj.get(zone)!;
+    const aabb = new Box3();
+    aabb.setFromObject(obj);
+    aabbs.set(zone, aabb);
+
+    const isDebugEnabled = APP.store.state.preferences.showAudioDebugPanel;
+    isDebugEnabled && !debugObjects.has(zone) && addZoneDebugObject(APP.world, zone);
+  });
+  audioZoneExitQuery(world).forEach(zone => {
+    aabbs.delete(zone);
+    debugObjects.delete(zone);
+
+    const isDebugEnabled = APP.store.state.preferences.showAudioDebugPanel;
+    isDebugEnabled && debugObjects.has(zone) && releaseZoneDebugObject(APP.world, zone);
+  });
+
+  const listener = APP.audioListener.eid!;
+  const listenersQuery = audioZoneListenerQuery(world);
```

Comment:
Do we ever expect more than one `AudioListenerTag`? If the expected result of `listenersQuery(world)` is `[listener]`, then perhaps we don't need the `AudioListenerTag` queries.

Reply:
The reason was more to keep track of the `AudioListener` in case it's removed or replaced in the scene. I don't thing there is any situation right now when we update the audio listener so maybe I was anticipating to something that might never happen.